### PR TITLE
Time Interval Solution Output

### PR DESCRIPTION
**Pull request by jbrillon**
PR contribution:
- Option to output solution files at set time intervals instead of strictly the iteration using `output_solution_every_dt_time_intervals` in `ode_solver/ode_solver_base.cpp` and `testing/flow_solver.cpp`

### CODE REVIEW

**Comment by dougshidong** [comment]
Before this becomes bigger. I just want to let you know that it's perfectly fine to have a PR for only outputting solution every dt intervals. As a reviewer, it's super easy to see that specific change and how it wouldn't break the code.

Up to you to add more code to your PR.

**Comment by dougshidong** [comment]
Also adding @PranshulThakur to every PR for when this will be open. I will slowly phase out my review comments, such that Donovan and Pranshul become your reviewers.

**Comment by jbrillon** [comment]
@dougshidong Ok good idea, I will open the PR so this can be reviewed and merged.

**Review by donovan97 — COMMENTED** [review]
Looks good to me.

**Review by PranshulThakur — COMMENTED** [review]
Great work!

File: src/ode_solver/ode_solver_base.h [hunk 1]
```diff
@@ -36,6 +36,11 @@ class ODESolverBase
     /** This variable will change when step_in_time() is called. */
     unsigned int current_iteration;
 
+    /// Current desired time for output solution every dt time intervals
+    /** This variable will change when advance_solution_time() is called
+     *  if ODE parameter "output_solution_every_dt_time_intervals" > 0. */
+    double current_desired_time_for_output_solution_every_dt_time_intervals;
```
[the remark this answers is not in the document]
Comment (jbrillon) [reply src/ode_solver/ode_solver_base.h:42]: I think that might be a bit vague given this is only for when you set the parameter `output_solution_every_dt_time_intervals` to non-zero positive. That name could be confusing when setting `output_solution_every_x_steps`.

[the remark this answers is not in the document]
Comment (PranshulThakur) [reply src/ode_solver/ode_solver_base.h:42]: Yeah, probably. Ok, I see. So it's basically `current_desired_time_for_parameter`.  Sure, that looks good. Let's keep it that way.

**Review by PranshulThakur — APPROVED** [review]

**Review by donovan97 — APPROVED** [review]

**Comment by dougshidong** [comment]
There is now a conflict since I merged Donovan's PR #88

**Pull request closed without merging**